tests/integration/test_charm_with_profile.py

```
@@ -0,0 +1,135 @@
+        entity_url="kubeflow-roles",
+        channel="latest/edge",
+        trust=True,
+    )
+    await ops_test.model.deploy(
+        entity_url="kubeflow-profiles",
+        channel="latest/edge",
+        trust=True,
+    )
+
+    await ops_test.model.wait_for_idle(
+        status="active", raise_on_blocked=True, raise_on_error=True, timeout=60 * 10
+    )
+
+
+@pytest.mark.parametrize("example", glob.glob("examples/*.yaml"))
+@pytest.mark.abort_on_fail
+async def test_authorization_for_creating_resources(
+    example, ops_test: OpsTest, lightkube_client, apply_profile
+):
+    """Assert a *Job can be created by a user in the user namespace."""
+    # Set up for creating an object of kind *Job
+    job_yaml = yaml.safe_load(Path(example).read_text())
+    training_job = job_yaml["kind"]
+    _, stdout, __ = await ops_test.run(
+        "kubectl",
+        "auth",
+        "can-i",
+        "create",
+        f"{training_job}",
+        f"--as=system:serviceaccount:{PROFILE_NAMESPACE}:default-editor",
+        f"--namespace={PROFILE_NAMESPACE}",
+        check=True,
+        fail_msg="Failed to execute kubectl auth",
+    )
+    assert stdout.strip() == "yes"
+
+
+def apply_manifests(lightkube_client: lightkube.Client, yaml_file_path: str):
+    """Apply resources using manifest files and returns the applied object.
+
+    Args:
+        lightkube_client (lightkube.Client): an instance of lightkube.Client to
+            use for applying resources.
+        yaml_file_path (str): the resource yaml file.
+
+    Returns:
+        A namespaced or global lightkube resource (obj).
+    """
+    read_yaml = Path(yaml_file_path).read_text()
+    yaml_loaded = codecs.load_all_yaml(read_yaml)
+    for obj in yaml_loaded:
+        try:
+            lightkube_client.apply(
+                obj=obj,
+                name=obj.metadata.name,
+            )
+        except lightkube.core.exceptions.ApiError as api_error:
+            raise api_error
```

Since we're not doing anything with this error, can't we just let it be raised?
```suggestion
        lightkube_client.apply(
            obj=obj,
            name=obj.metadata.name,
        )
```